hoomd/mpcd/collide.py

```diff
@@ -1,367 +1,225 @@
+            each edge length a multiple of `cell_size`.
 
-        Args:
-            group (``hoomd.group``): Group of particles to embed
+        shift (bool): When True, randomly shift underlying collision cells.
 
-        The *group* is embedded into the MPCD collision step and cell properties.
-        During collisions, the embedded particles are included in determining
-        per-cell quantities, and the collisions are applied to the embedded
-        particles.
+            When the total mean-free path of the MPCD particles is small, the
+            underlying MPCD cell list should be randomly shifted in order to
+            ensure Galilean invariance. The performance penalty from grid
+            shifting is small, so it is recommended to enable it in all
+            simulations.
 
-        No integrator is generated for *group*. Usually, you will need to create
-        a separate method to integrate the embedded particles. The recommended
-        (and most common) integrator to use is :py:class:`~hoomd.md.methods.NVE`.
-        It is generally **not** a good idea to use a thermostatting integrator for
-        the embedded particles, since the MPCD particles themselves already act
-        as a heat bath that will thermalize the embedded particles.
-
-        Examples::
-
-            polymer = hoomd.group.type('P')
-            md.integrate.nve(group=polymer)
-            method.embed(polymer)
-
-        """
-
-        self.group = group
-        self._cpp.setEmbeddedGroup(group.cpp_group)
-
-    def enable(self):
-        """ Enable the collision method
+    """
 
-        Examples::
+    def __init__(self, cell_size, shift=True):
+        super().__init__()
 
-            method.enable()
+        param_dict = ParameterDict(
+            cell_size=float(cell_size),
+            shift=bool(shift),
+        )
+        self._param_dict.update(param_dict)
 
-        Enabling the collision method adds it to the current MPCD system definition.
-        Only one collision method can be attached to the system at any time.
-        If another method is already set, ``disable`` must be called
-        first before switching.
+    def _attach_hook(self):
+        sim = self._simulation
+        if isinstance(sim.device, hoomd.device.GPU):
+            cpp_class = _mpcd.CellListGPU
+        else:
+            cpp_class = _mpcd.CellList
 
-        """
+        self._cpp_obj = cpp_class(sim.state._cpp_sys_def, self.cell_size,
+                                  self.shift)
 
-        self.enabled = True
-        hoomd.context.current.mpcd._collide = self
+        super()._attach_hook()
 
-    def disable(self):
-        """ Disable the collision method
 
-        Examples::
+class CollisionMethod(AutotunedObject):
+    """Base collision method.
 
-            method.disable()
+    Args:
+        every (int): Number of integration steps between collisions.
+        embed (hoomd.filter.filter_like): HOOMD particles to include in collision.
```

Comment:
I considered `filter` but decided against it because I thought it sounded like the method would only be applied to the particles in `filter`, and not also the MPCD particles. How about `embedded_particles` in keeping with Josh's preference for nouns?